### PR TITLE
Support timezone in csv header and refactored header parsing for points

### PR DESCRIPTION
**Pull request by SaschaPeukert**

### CODE REVIEW

File: community/csv/src/main/java/org/neo4j/csv/reader/Extractors.java [hunk 1]
```diff
@@ -906,7 +907,7 @@ protected void extract0( char[] data, int offset, int length, AnyValue[] optiona
         }
 
         @Override
-        protected void extract0( char[] data, int offset, int length, AnyValue[] optionalDatah )
+        protected void extract0( char[] data, int offset, int length, CSVHeaderInformation optionalDatah )
```
Comment (Lojjs) [inline community/csv/src/main/java/org/neo4j/csv/reader/Extractors.java]: Why is this one called `optionalDatah` when all the others are called `optionalData`, typo?

File: community/csv/src/main/java/org/neo4j/csv/reader/Extractors.java [hunk 2]
```diff
@@ -1062,14 +1064,14 @@ public AnyValue value()
 
         TimeExtractor( Supplier<ZoneId> defaultTimeZone )
         {
-            super( "Time" );
+            super( NAME );
```
Comment (Lojjs) [inline community/csv/src/main/java/org/neo4j/csv/reader/Extractors.java:1069]: For consistency, you maybe could consider to do this refactoring for the remaining temporal types as well, even if their `NAME` will only be used in one place for now.

File: community/values/src/main/java/org/neo4j/values/storable/PointValue.java [hunk 3]
```diff
@@ -264,153 +261,64 @@ public boolean withinRange( PointValue lower, boolean includeLower, PointValue u
...
     public static PointValue fromMap( MapValue map )
     {
-        AnyValue[] fields = new Value[PointValueField.values().length];
-        for ( PointValueField f : PointValueField.values() )
+        PointCSVHeaderInformation fields = new PointCSVHeaderInformation();
+        for ( Map.Entry<String,AnyValue> entry : map.entrySet() )
         {
-            AnyValue fieldValue = map.get( f.name().toLowerCase() );
-            fields[f.ordinal()] = fieldValue != Values.NO_VALUE ? fieldValue : null;
+            fields.assign( entry.getKey().toLowerCase(), entry.getValue() );
```
Comment (Lojjs) [inline community/values/src/main/java/org/neo4j/values/storable/PointValue.java:267]: Before, you had some converting between `NO_VALUE` and `null` here. Can we never get `NO_VALUE` in the current solution?

**Review by Lojjs — COMMENTED** [review]
Overall it looks good, I just had some small comments. If it is something you want to fix, you can choose between doing it here or in #11305

File: community/values/src/main/java/org/neo4j/values/storable/PointValue.java (hunk 3 above)
Comment (SaschaPeukert) [reply community/values/src/main/java/org/neo4j/values/storable/PointValue.java:267]: We might get a ```NO_VALUE```, but ```assign()``` would throw an exception if we encounter it (would be the same if we pass in null, so it makes no difference to check for ```NO_VALUE```

**Review by Lojjs — APPROVED** [review]

**Pull request merged**